### ISSUE TITLE
Invalid type cycles in regex

### ISSUE THREAD

**Issue by heshanpadmasiri** [ballerina-platform/nballerina#1088]
Nesting a star inside a star (`(a*)*`) produces an invalid type cycle as
```
type T1 T2 | T0;
type T2 T4 | T1;
type T4 [T3, T2];
type T3 "a";
type T0 ();

```

**Comment by heshanpadmasiri** [comment]
Reopening the issue since I found another case with the same bug that was not solved by the earlier fix #1089. 
```
// (a|((b|c)*))*

type T1 T2 | T0;
type T2 T4 | T5;
type T4 [T3, T1];
type T3 "a";
type T5 T6 | T1;
type T6 T8 | T10;
type T8 [T7, T5];
type T7 "b";
type T10 [T9, T5];
type T9 "c";
type T0 ();
```

The root cause of the problem seems to be when we have a nested star or a union including a star inside a star, the type definition of the inner star will include an element that refers to the outer star. Since stars are represented as unions this ultimately leads to two unions that refer to each other.

**Comment by lochana-chathura** [comment]
The test case: 
`["<", "(RgeRgF(b((OS)*cDVB)*H)*kYLy)*W", "(RgeRgF(b(((glPrQ)*OS)*cDVB)*H)*kYLy)*W"]`

added with PR #1089 failing intermittently with StackOverflow after integrating readonly cells for regex in PR #1091.
I have temporary disabled it with https://github.com/ballerina-platform/nballerina/pull/1091/commits/8e939fcca77074873bdfecfd3208cb0f65428ae8
